pandas/core/algorithms.py

```diff
@@ -633,6 +643,13 @@ def factorize(
     values = _ensure_arraylike(values)
     original = values
 
+    # GH35667, if na_sentinel=None, we will not dropna NaNs from the uniques
+    # of values, still assign na_sentinel=-1 to keep backward compatability.
```

Comment:
> still assign na_sentinel=-1 to keep backward compatability.

That's not for backwards compatibility AFAIU, but rather so that those can be replaced with the code value for NaN (code on lines 674-680 below)?

Reply:
yep, rephrased!